docs/pages/docs/core-concepts/caching.mdx

```diff
@@ -7,9 +7,40 @@ import Callout from "../../../components/Callout";
 
 # Caching
 
-Unlike other tools you may have used, Turborepo can cache emitted files and logs of previously run commands, so it can skip work that has already been done, yielding incredible time savings.
+Turborepo ships with a **task cache**. It can cache the results of any task - meaning you never need to do the same work twice.
 
-By default, [`turbo run`](../reference/command-line-reference#turbo-run-task1-task2-1) considers any files emitted in the `dist/**` or `build/**` folders of a workspace task (defined in that workspace's `package.json` `scripts` object) to be cacheable. In addition, `turbo run` treats logs (`stderr` and`stdout`), which are automatically written to `.turbo/run-<command>.log`, as a cacheable artifact. By treating logs as artifacts, you can cache just about any command in your Turborepo.
+## Missing the cache
+
+Each task in your monorepo has an **input** and an **output**.
+
+- A `build` task takes in **source files as input** and returns **bundled files as output**.
+- A `lint` or `test` task also takes in **source files**, but outputs to **`stdout`**.
+
+Let's say you run a `build` task with Turborepo using `turbo run build`:
+
+![](../../../public/images/docs/cache-miss.png)
+
+1. Turborepo will **take the inputs to your task** (by default all non-gitignored files), and **turn them into a hash**, like `78awdk123`.
+
+2. It'll then **check its local cache** for a folder named `78awdk123`.
```

Comment:
It checks all caches, local first and then remote if available.

Reply:
The user doesn't know about remote caching at this point.